administrator/components/com_fields/src/Helper/FieldsHelper.php

```diff
@@ -108,6 +108,15 @@ public static function getFields($context, $item = null, $prepareValue = false,
 			self::$fieldsCache->setState('list.limit', 0);
 		}
 
+		if (!$includeSubformFields)
+		{
+			self::$fieldsCache->setState('filter.only_use_in_subform', 0);
+		}
+		else
+		{
+			self::$fieldsCache->setState('filter.only_use_in_subform', '');
+		}
```

Comment:
```suggestion
		if ($includeSubformFields)
		{
			self::$fieldsCache->setState('filter.only_use_in_subform', '');
		}
		else
		{
			self::$fieldsCache->setState('filter.only_use_in_subform', 0);
		}
```
Couldn't we use positive logic here?

Reply:
Either way works for me 🤷🏽‍♂️

Reply:
It's nicer with positive logic according to school books ;-)

Reply:
I prefer to have the code that applies the non–default filter first. This makes it far less likely that someone will see it and go “oh, this parameter and if–block are redundant, we are just setting the filter to its default value”, remove it and break stuff because they didn't actually read the code or run a git blame to see WTF this code is there. It may or may not have happened before...

Reply:
BTW I did merge your suggestion or at least I told GitHub to do so.

Reply:
Yes, you merged it. But if you think it was wrong and your original code was better to understand or to read, then change it back. It was just a suggestion, and as a human being I might be wrong with a suggestion sometimes. It's just school books say positive logic shall be preferred because normally that's better readable.

Reply:
Nah, it's all good. Let's keep it the way you proposed. It's probably more readable for people who are not like me.

Being neurodiverse I read code differently than most people. I don't care about inverted or negative logic. Same for double negatives. They don't trip me over. What trips me over to some extent is if the “interesting“ branch of an if–block is not at the top. It slows me down reading that code. I also recognise that this is NOT how most people read code — I taught other people to code about a decade ago, a very informing and humbling experience for me. So, when it's not code that I am most likely to mostly read and work on myself I am happy to accommodate the majority of people who will be reading and working on it. In the end of the day FOSS is about helping _other people_ with our code, right? 😄